src/Haskell/Ide/Engine/Plugin/Ormolu.hs

```diff
@@ -0,0 +1,38 @@
+{-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE OverloadedStrings #-}
+
+module Haskell.Ide.Engine.Plugin.Ormolu ( ormoluDescriptor ) where
+
+import Control.Monad.IO.Class ( liftIO, MonadIO(..) )
+import Control.Exception
+import Data.Aeson ( Value ( Null ) )
+import qualified Data.Text as T
+import Ormolu
+import Ormolu.Config (defaultConfig)
+import Ormolu.Exception (OrmoluException)
+import Haskell.Ide.Engine.MonadTypes
+import Haskell.Ide.Engine.PluginUtils
+
+ormoluDescriptor :: PluginId -> PluginDescriptor
+ormoluDescriptor plId = PluginDescriptor
+  { pluginId                 = plId
+  , pluginName               = "Ormolu"
+  , pluginDesc               = "A formatter for Haskell source code."
+  , pluginCommands           = []
+  , pluginCodeActionProvider = Nothing
+  , pluginDiagnosticProvider = Nothing
+  , pluginHoverProvider      = Nothing
+  , pluginSymbolProvider     = Nothing
+  , pluginFormattingProvider = Just provider
+  }
+
+provider :: FormattingProvider
+provider contents uri typ _opts =
+  case typ of 
+    -- // TODO Adequately throw an error on the following line
+    FormatRange _ -> return $ IdeResultFail (IdeError PluginError (T.pack $  "ormoluCmd: Selection formatting not supported.") Null)
```

Comment:
Just from looking at the Ormolu API, I think we should be able to support range formatting! The file path is only used "to use it in parse error messages" according to the haddock, so we should be able to extract part of the text and feed it into `ormolu`
You should try the `extractRange` helper function that the other formatting providers use:

https://github.com/haskell/haskell-ide-engine/blob/b8c99d173f8062e633f95370423d7237956798ab/src/Haskell/Ide/Engine/Plugin/Floskell.hs#L37-L39

Reply:
Just was pointed out that apparently ormolu doesn't support fragments this way. This is fine then! We can add this if ormolu ever decides to support it